samples/gallery/src/main/java/com/bumptech/glide/samples/gallery/HorizontalGalleryFragment.kt

```diff
@@ -1,50 +1,559 @@
+    DeviceMedia(mediaStoreData.value)
+  }
+
+  @Composable
+  fun DeviceMedia(mediaStoreData: List<MediaStoreData>) {
+    val state = rememberLazyListState()
+    LazyRow(horizontalArrangement = Arrangement.spacedBy(10.dp), state = state) {
+      items(mediaStoreData) { mediaStoreItem ->
+        MediaStoreView(mediaStoreItem, Modifier.fillParentMaxSize())
+      }
+    }
+
+    GlideLazyListPreloader(
+      state = state,
+      data = mediaStoreData,
+      size = Size(50f, 50f),
+      numberOfItemsToPreload = 15,
+      fixedVisibleItemCount = 2,
+    ) { data, requestBuilder -> requestBuilder.load(data.uri).signature(data.signature()) }
+  }
+
+  private fun MediaStoreData.signature() =
+    MediaStoreSignature(mimeType, dateModified, orientation)
+
+  @Composable
+  fun MediaStoreView(item: MediaStoreData, modifier: Modifier) {
+    val requestManager = Glide.with(requireContext())
+    val signature = item.signature()
+
+    GlideImage(
+      requestManager = requestManager,
+      model = item.uri,
+      requestBuilder = requestManager
+        .asDrawable()
+        .load(item.uri)
+        .thumbnail(
+          requestManager.asDrawable().load(item.uri).signature(signature).override(50))
+        .signature(signature),
+      ImageParameters(contentDescription = item.displayName, modifier = modifier)
+    )
+  }
+
+  // TODO(judds): Everything below this line belongs in Glide, not in a sample app.
+
+  /**
+   * Preloads ahead of the users current scroll position for [LazyRow] and
+   * [androidx.compose.foundation.lazy.LazyColumn], similar to [ListPreloader] and
+   * [com.bumptech.glide.integration.recyclerview.RecyclerViewPreloader].
+   *
+   * @param state The [LazyListState] provided to the `LazyRow` or `LazyColumn`
+   * @param data The backing list of metadata that we're going to preload images for.
+   * @param size The override size we'll pass to [RequestBuilder.override] .
+   * @param numberOfItemsToPreload The number of items to preload ahead of the user's current
+   * position. This should be tested for each application. If the total memory size of the preloaded
+   * images exceeds the memory cache, preloading is effectively useless. However if you preload too
+   * few things, the buffer may be small enough that images are not available when they could be, so
+   * it's always a balancing act. The smaller the preloaded image, the more you can preload.
+   * @param fixedVisibleItemCount The number of visible items. In some cases this can vary widely
+   * in which case you can leave this value `null`. If the number of visible items is always one
+   * or two, it might make sense to just set this to the larger of the two to reduce churn in the
+   * preloader.
+   * @param requestBuilderModifier See [ListPreloader.PreloadModelProvider.getPreloadRequestBuilder]
+   */
+  // TODO(judds): Consider wrapping a LazyRow / LazyColumn and providing state instead of a separate
```

Comment:
++ I think this would be a better API as well. Though I think it's good to also have an API that is not tied to LazyRow & LazyColumn for usages outside of those contexts.